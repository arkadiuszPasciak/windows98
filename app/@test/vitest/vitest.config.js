import { defineProject, mergeConfig } from "vitest/config"
import sharedConfig from "@windows98/configs/vitest/shared"
import tsconfigPaths from "vite-tsconfig-paths"

export default mergeConfig(
	sharedConfig,
	defineProject({
		plugins: [tsconfigPaths({ loose: true })],
	})
)
