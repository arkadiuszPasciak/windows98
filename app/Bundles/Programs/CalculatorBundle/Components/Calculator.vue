<template>
  <UIModal
    class="Calculator"
    :title="t('CalculatorBundle.title')"
    :resize-window="false"
    width="195px"
    height="220px"
    :modal-state="programStore.modalCalculator"
    @close-modal="closeModal"
  >
    <div class="content">
      <CalculatorDisplay />
      <CalculatorKeyboard />
    </div>
  </UIModal>
</template>

<script setup lang="ts">
  import { useI18n } from 'vue-i18n'
  import CalculatorDisplay from '@APP|Bundles/CalculatorBundle/Components/CalculatorDisplay.vue'
  import CalculatorKeyboard from '@APP|Bundles/CalculatorBundle/Components/CalculatorKeyboard.vue'
  import UIModal from '@APP|Bundles/UIModalBundle/Components/UIModal.vue'
  import { useProgramStore } from '@APP|Bundles/ProgramBundle/Stores/Program.stores'

  const programStore = useProgramStore()

  const closeModal = (): void => {
    programStore.updateCalculatorModal(false)
  }

  const { t } = useI18n()
</script>

<style
  lang="scss"
  scoped
  src="@APP|Bundles/CalculatorBundle/Styles/Calculator.styles.scss"
/>
