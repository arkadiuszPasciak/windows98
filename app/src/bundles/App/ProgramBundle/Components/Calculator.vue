<template>
  <UIModal
    class="Calculator"
    :title="t('CalculatorBundle.title')"
    :resize-window="false"
    width="195px"
    height="220px"
    :modal-state="programStore.modalCalculator"
    @close-modal="closeModal"
  >
    <ma-calculator />
  </UIModal>
</template>

<script setup lang="ts">
import { useProgramStore } from "@APP/src/bundles/App/ProgramBundle/Stores/Program.stores"
import UIModal from "@APP|Bundles/UIModalBundle/Components/UIModal.vue"
import { useI18n } from "vue-i18n"

const programStore = useProgramStore()

const closeModal = (): void => {
	programStore.updateCalculatorModal(false)
}

const { t } = useI18n()
</script>
