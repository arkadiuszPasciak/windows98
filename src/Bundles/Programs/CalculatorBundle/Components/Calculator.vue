<template>
  <UIModal
    class="Calculator"
    :title="t('CalculatorBundle.title')"
    :resize-window="false"
    width="195px"
    height="220px"
    :modal-state="programStore.modalCalculator"
    @close-modal="closeModal"
  >
    <div class="content">
      <CalculatorDisplay />
      <CalculatorKeyboard />
    </div>
  </UIModal>
</template>

<script setup lang="ts">
  import { useI18n } from 'vue-i18n'
  import CalculatorDisplay from '@Bundles/CalculatorBundle/Components/CalculatorDisplay.vue'
  import CalculatorKeyboard from '@Bundles/CalculatorBundle/Components/CalculatorKeyboard.vue'
  import UIModal from '@Bundles/UIBundle/Components/UIModal.vue'
  import { useProgramStore } from '@Bundles/ProgramBundle/Stores/Program.stores'

  const programStore = useProgramStore()

  const closeModal = (): void => {
    programStore.updateCalculatorModal(false)
  }

  const { t } = useI18n()
</script>

<i18n src="@Bundles/CalculatorBundle/Locales/Calculator.locales.json"></i18n>

<style
  lang="scss"
  scoped
  src="@Bundles/CalculatorBundle/Styles/Calculator.styles.scss"
/>
