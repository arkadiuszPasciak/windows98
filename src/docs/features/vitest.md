# [Vitest](https://vitest.dev/)

## Description

Vitest is a modern and fast testing framework specifically designed for Vite-powered projects. Developed by the [Vite team](https://vitejs.dev/team.html), Vitest leverages Vite's architecture to provide a seamless testing experience with features like native ESM support, rapid test execution, and isolated testing environments. It offers compatibility with various front-end libraries and frameworks, such as Vue, React, and Svelte. With its straightforward configuration and integration with Vite, Vitest aims to enhance the developer experience by providing an efficient and enjoyable testing workflow for Vite-based web applications.

## Commands

[Available commands in the application](/reference/cli.html#vitest)
