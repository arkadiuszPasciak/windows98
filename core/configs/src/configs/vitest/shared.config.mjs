import { defineConfig } from 'vitest/config'

export default defineConfig({
    test: {
        environment: "jsdom",
        exclude: [
            "**/build/**",
            "**/dist/**",
            "**/node_modules/**",
            "**/@bundler/**",
            "**/@linter/**",
            "**/@test/**",
        ],
        include: [
            "**/*.unit.ts"
        ],
        globals: true,
    },
})
