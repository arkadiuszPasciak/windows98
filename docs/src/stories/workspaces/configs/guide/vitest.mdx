import { Meta, Title } from "@storybook/addon-docs/blocks";

<Meta title="Workspaces/Configs/Guide/Vitest" />

<Title>âš¡ Vitest</Title>

Complete guide to using Vitest configuration from `@windows98/configs` for unit and integration testing.

## Overview

Vitest configuration provides fast, modern testing setup optimized for TypeScript and modern JavaScript frameworks within the Windows98 ecosystem.

## Configuration Path

```
@windows98/configs/vitest/shared
```

### Install Dependencies

```bash
# Install vitest
pnpm add -D vitest

# Install the configs package
pnpm add -D @windows98/configs
```

## Basic Usage

### Simple Setup

```javascript
// ./@test/vitest/vitest.config.js
import { defineConfig } from 'vitest/config'
import sharedConfig from '@windows98/configs/vitest/shared'

export default defineConfig(sharedConfig)
```

### With Custom Settings

```javascript
// ./@test/vitest/vitest.config.js
import { defineConfig } from 'vitest/config'
import sharedConfig from '@windows98/configs/vitest/shared'

export default defineConfig({
  ...sharedConfig,
  test: {
    ...sharedConfig.test,
    environment: 'jsdom',
    setupFiles: ['./src/test-setup.ts']
  }
})
```

## Package.json Scripts

```json
{
  "scripts": {
	"test:unit:run": "vitest run -c ./@test/vitest/vitest.config.js",
	"test:unit:watch": "vitest -c ./@test/vitest/vitest.config.js",
  }
}
```

