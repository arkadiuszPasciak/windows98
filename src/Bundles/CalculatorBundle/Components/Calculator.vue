<template>
  <UIModal
    class="Calculator"
    :title="t('Calculator.title')"
    :resize-window="false"
    :width="195"
    :height="220"
    :modal-state="programStore.modalCalculator"
    @close-modal="closeModal"
  >
    <div class="content">
      <CalculatorDisplay class="display" :value="calculatorStore.value" />
      <UIButton
        class="color-blue"
        size="calculator"
        @click="addNumber(ECalculatorNumber.SEVEN)"
      >
        {{ ECalculatorNumber.SEVEN }}
      </UIButton>
      <UIButton
        class="color-blue"
        size="calculator"
        @click="addNumber(ECalculatorNumber.EIGHT)"
      >
        {{ ECalculatorNumber.EIGHT }}
      </UIButton>
      <UIButton
        class="color-blue"
        size="calculator"
        @click="addNumber(ECalculatorNumber.NINE)"
      >
        {{ ECalculatorNumber.NINE }}
      </UIButton>
      <UIButton
        class="color-red"
        size="calculator"
        @click="addSign(ECalculatorSign.DIVIDE)"
      >
        /
      </UIButton>
      <UIButton
        class="color-blue"
        size="calculator"
        @click="addNumber(ECalculatorNumber.FOUR)"
      >
        {{ ECalculatorNumber.FOUR }}
      </UIButton>
      <UIButton
        class="color-blue"
        size="calculator"
        @click="addNumber(ECalculatorNumber.FIVE)"
      >
        {{ ECalculatorNumber.FIVE }}
      </UIButton>
      <UIButton
        class="color-blue"
        size="calculator"
        @click="addNumber(ECalculatorNumber.SIX)"
      >
        {{ ECalculatorNumber.SIX }}
      </UIButton>
      <UIButton
        class="color-red"
        size="calculator"
        @click="addSign(ECalculatorSign.MULTIPLY)"
      >
        *
      </UIButton>
      <UIButton
        class="color-blue"
        size="calculator"
        @click="addNumber(ECalculatorNumber.ONE)"
      >
        {{ ECalculatorNumber.ONE }}
      </UIButton>
      <UIButton
        class="color-blue"
        size="calculator"
        @click="addNumber(ECalculatorNumber.TWO)"
      >
        {{ ECalculatorNumber.TWO }}
      </UIButton>
      <UIButton
        class="color-blue"
        size="calculator"
        @click="addNumber(ECalculatorNumber.THREE)"
      >
        {{ ECalculatorNumber.THREE }}
      </UIButton>
      <UIButton
        class="color-red"
        size="calculator"
        @click="addSign(ECalculatorSign.MINUS)"
      >
        -
      </UIButton>
      <UIButton
        class="color-blue"
        size="calculator"
        @click="addNumber(ECalculatorNumber.ZERO)"
      >
        {{ ECalculatorNumber.ZERO }}
      </UIButton>
      <UIButton class="color-blue" size="calculator" @click="addDot()">
        .
      </UIButton>
      <UIButton class="color-blue" size="calculator" @click="summResult()">
        =
      </UIButton>
      <UIButton
        class="color-red"
        size="calculator"
        @click="addSign(ECalculatorSign.ADD)"
      >
        +
      </UIButton>
    </div>
  </UIModal>
</template>

<script setup lang="ts">
  import { useI18n } from 'vue-i18n'
  import CalculatorDisplay from '@Bundles/CalculatorBundle/Components/CalculatorDisplay.vue'
  import UIButton from '@Bundles/UIBundle/Components/UIButton.vue'
  import UIModal from '@Bundles/UIBundle/Components/UIModal.vue'
  import { useCalculatorStore } from '@Bundles/CalculatorBundle/Stores/Calculator.stores'
  import { useProgramStore } from '@Bundles/ProgramBundle/Stores/Program.stores'
  import {
    ECalculatorNumber,
    TCalculatorNumber,
    ECalculatorSign,
    TCalculatorSign,
  } from '@Bundles/CalculatorBundle/Supports/Calculator.supports'

  const calculatorStore = useCalculatorStore()
  const programStore = useProgramStore()

  const closeModal = (): void => {
    programStore.updateCalculatorModal(false)
  }

  const addDot = (): void => {
    calculatorStore.addDot()
  }

  const addNumber = (number: TCalculatorNumber): void => {
    calculatorStore.addNumber(number)
  }

  const addSign = (sign: TCalculatorSign): void => {
    calculatorStore.addSign(sign)
  }

  const summResult = (): void => {
    calculatorStore.summResult()
  }

  const { t } = useI18n()
</script>

<i18n src="@Bundles/CalculatorBundle/Locales/Calculator.locales.json"></i18n>

<style lang="scss" scoped>
  @import '@Bundles/CalculatorBundle/Styles/Calculator.styles.scss';
</style>
