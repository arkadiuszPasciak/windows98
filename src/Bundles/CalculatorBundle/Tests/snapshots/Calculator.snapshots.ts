import { mount } from '@vue/test-utils'
import { expect, it, describe } from 'vitest'
import Calculator from '@Bundles/CalculatorBundle/Components/Calculator.vue'

const defaultWrapper = mount(Calculator)

describe('[CalculatorBundle]<Snapshots>(Calculator)', async () => {
  it('renders correctly component', () => {
    expect(defaultWrapper.element).toMatchSnapshot()
  })
})
