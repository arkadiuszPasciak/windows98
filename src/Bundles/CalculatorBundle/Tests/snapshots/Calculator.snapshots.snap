// Vitest Snapshot v1

exports[`[CalculatorBundle]<Snapshots>(Calculator) > renders correctly component 1`] = `
<div
  class="UIModal move-window Calculator"
  data-v-99a75817=""
  data-v-a7eaed9a=""
>
  <div
    class="container"
    data-v-99a75817=""
    style="width: 195px; height: 220px;"
  >
    <div
      class="header"
      data-v-99a75817=""
      style="cursor: default;"
    >
      <h3
        class="title"
        data-v-99a75817=""
      >
        Calculator
      </h3>
      <button
        class="UIButton size-small button-close"
        data-v-99a75817=""
        data-v-cf1bb3a9=""
        type="button"
      >
        
        <img
          alt="Close window"
          class="button-close-icon"
          data-v-99a75817=""
          height="7"
          src="/icons/close-modal.svg"
          width="8"
        />
        
        
        
        
        
      </button>
    </div>
    <!--v-if-->
    
    <div
      class="content"
      data-v-a7eaed9a=""
    >
      <p
        class="UIText CalculatorDisplay display"
        data-v-a7eaed9a=""
        data-v-e69f73e7=""
        data-v-f0b878d5=""
      >
        
        0
        
      </p>
      <button
        class="UIButton size-calculator color-blue"
        data-test="calculator-button-7"
        data-v-a7eaed9a=""
        data-v-cf1bb3a9=""
        type="button"
      >
        
        
        
        7
        
        
        
      </button>
      <button
        class="UIButton size-calculator color-blue"
        data-v-a7eaed9a=""
        data-v-cf1bb3a9=""
        type="button"
      >
        
        
        
        8
        
        
        
      </button>
      <button
        class="UIButton size-calculator color-blue"
        data-v-a7eaed9a=""
        data-v-cf1bb3a9=""
        type="button"
      >
        
        
        
        9
        
        
        
      </button>
      <button
        class="UIButton size-calculator color-red"
        data-v-a7eaed9a=""
        data-v-cf1bb3a9=""
        type="button"
      >
        
        
        
         / 
        
        
        
      </button>
      <button
        class="UIButton size-calculator color-blue"
        data-v-a7eaed9a=""
        data-v-cf1bb3a9=""
        type="button"
      >
        
        
        
        4
        
        
        
      </button>
      <button
        class="UIButton size-calculator color-blue"
        data-v-a7eaed9a=""
        data-v-cf1bb3a9=""
        type="button"
      >
        
        
        
        5
        
        
        
      </button>
      <button
        class="UIButton size-calculator color-blue"
        data-v-a7eaed9a=""
        data-v-cf1bb3a9=""
        type="button"
      >
        
        
        
        6
        
        
        
      </button>
      <button
        class="UIButton size-calculator color-red"
        data-v-a7eaed9a=""
        data-v-cf1bb3a9=""
        type="button"
      >
        
        
        
         * 
        
        
        
      </button>
      <button
        class="UIButton size-calculator color-blue"
        data-v-a7eaed9a=""
        data-v-cf1bb3a9=""
        type="button"
      >
        
        
        
        1
        
        
        
      </button>
      <button
        class="UIButton size-calculator color-blue"
        data-v-a7eaed9a=""
        data-v-cf1bb3a9=""
        type="button"
      >
        
        
        
        2
        
        
        
      </button>
      <button
        class="UIButton size-calculator color-blue"
        data-v-a7eaed9a=""
        data-v-cf1bb3a9=""
        type="button"
      >
        
        
        
        3
        
        
        
      </button>
      <button
        class="UIButton size-calculator color-red"
        data-v-a7eaed9a=""
        data-v-cf1bb3a9=""
        type="button"
      >
        
        
        
         - 
        
        
        
      </button>
      <button
        class="UIButton size-calculator color-blue"
        data-v-a7eaed9a=""
        data-v-cf1bb3a9=""
        type="button"
      >
        
        
        
        0
        
        
        
      </button>
      <button
        class="UIButton size-calculator color-blue"
        data-v-a7eaed9a=""
        data-v-cf1bb3a9=""
        type="button"
      >
        
        
        
         . 
        
        
        
      </button>
      <button
        class="UIButton size-calculator color-blue"
        data-v-a7eaed9a=""
        data-v-cf1bb3a9=""
        type="button"
      >
        
        
        
         = 
        
        
        
      </button>
      <button
        class="UIButton size-calculator color-red"
        data-v-a7eaed9a=""
        data-v-cf1bb3a9=""
        type="button"
      >
        
        
        
         + 
        
        
        
      </button>
    </div>
    
  </div>
</div>
`;
